db/compaction_job.cc

```diff
@@ -1498,7 +1498,7 @@ Status CompactionJob::OpenCompactionOutputFile(
       *cfd->ioptions(), *(sub_compact->compaction->mutable_cf_options()),
       cfd->internal_comparator(), cfd->int_tbl_prop_collector_factories(),
       cfd->GetID(), cfd->GetName(), sub_compact->outfile.get(),
-      sub_compact->compaction->output_compression(),
+      sub_compact->compaction->output_compression(), 0 /*sample_for_compression */,
```

Comment:
After rethinking about this, I think it's easy to replace the 0 with the following:
```
bottommost_level_ ? sub_compact->compaction->mutable_cf_options()->sample_for_compression : 0
```
Therefore we can sample the compressibility for bottommost level compaction if the user enables it.

Reply:
Let's address this later.